src/ui/layout/box-constraints.md

```diff
@@ -1,133 +0,0 @@
----
```

Comment:
Overall, the restructuring looks good to me. There is one concern I have, though: This page was linked to from a very specific error message in the framework [1] and offered a very tailored description of that problem. What link do we provide in the error message after this restructuring? If we just link to common-errors it may degrade the user experience as now I have to figure out manually which of the common error scenarios there apply to me.


[1] https://github.com/flutter/flutter/blob/c8e42b47f5ea8b5ff7bf2f2b0a2a8e765f1aa51d/packages/flutter/lib/src/rendering/flex.dart#L733

Reply:
So, it looks like it specifically links to that page for the unbounded error? Is that right?

Reply:
Yes!

Reply:
With these latest changes do we need to change the link in the framework to `ui/layout/box-constraints`?

Reply:
Actually, that doesn't seem right. `ui/layout/box-constraints` talks about unbounded scrollables, whereas the error in https://github.com/flutter/flutter/blob/c8e42b47f5ea8b5ff7bf2f2b0a2a8e765f1aa51d/packages/flutter/lib/src/rendering/flex.dart#L733 is about unbounded constraints in flex widgets.... So, not sure what to do here...

Reply:
Hi, @goderbauer! OK, I carefully read through the original box constraints page and the two sections I've added to this doc: the unbounded constraints section and the Flex section directly below that. It essentially says the exact same thing as the two sections in Ian's page.

So, what I did to make this clearer, is to add a note to the beginning of the Unbounded constraints page to say that the tooling might direct you here, but also read the Flex section below. What do you think?